bluemira/neutronics/params.py

```diff
@@ -0,0 +1,309 @@
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""dataclasses containing parameters used to set up the openmc model."""
+
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import Literal, Tuple, Union
+
+import openmc
+
+from bluemira.base.constants import raw_uc
+from bluemira.neutronics.constants import energy_per_dt
+from bluemira.neutronics.make_materials import BlanketType
+
+
+@dataclass
+class OpenMCSimulationRuntimeParameters:
+    """Parameters used in the actual simulation
+
+    Parameters
+    ----------
+    particles:
+        Number of neutrons emitted by the plasma source per batch.
+    batches:
+        How many batches to simulate.
+    photon_transport:
+        Whether to simulate the transport of photons (i.e. gamma-rays created) or not.
+    electron_treatment:
+        The way in which OpenMC handles secondary charged particles.
+        'thick-target bremsstrahlung' or 'local energy deposition'
+        'thick-target bremsstrahlung' accounts for the energy carried away by
+        bremsstrahlung photons and deposited elsewhere, whereas 'local energy
+        deposition' assumes electrons deposit all energies locally.
+        (the latter is expected to be computationally faster.)
+    run_mode:
+        see below for details:
+        https://docs.openmc.org/en/stable/usersguide/settings.html#run-modes
+    openmc_write_summary:
+        whether openmc should write a 'summary.h5' file or not.
+    cross_section_xml:
+        Where the xml file for cross-section is stored locally.
+    """
+
+    # Parameters used outside of setup_openmc()
+    parametric_source: bool  # to use the pps_isotropic module or not.
+    particles: int  # number of particles used in the neutronics simulation
+    cross_section_xml: Union[str, Path]
+    batches: int = 2
+    photon_transport: bool = True
+    electron_treatment: Literal[
+        "ttb", "led"
+    ] = "led"  # Bremsstrahlung only matters for very thin objects
+    run_mode: str = openmc.settings.RunMode.FIXED_SOURCE.value
+    openmc_write_summary: bool = False
+    # number of particles used in the volume calculation.
+    volume_calc_particles: int = int(4e8)
+
+
+@dataclass
+class BreederTypeParameters:
+    """Dataclass to hold information about the breeder blanket material
+    and design choices.
+    """
+
+    enrichment_fraction_Li6: float
+    blanket_type: BlanketType
+
+
+@dataclass(frozen=True)
+class TokamakOperationParameters:
+    """Parameters describing how the tokamak is operated,
+    i.e. where the plasma is positioned (and therefore where the power is concentrated),
+    and what temperature the plasma is at.
+
+    Parameters
+    ----------
+    reactor_power:
+        total reactor (thermal) power when operating at 100%
+    peaking_factor:
```

Comment:
The plasma neutron source we have requires its user to directly specify the number of neutrons coming out of the plasma, rather than use the plasma densities to calculate the number of neutrons (which can then be used to calculate the reactor power, as you have pointed out).

neutronics.ex.py assumes all neutrons produced are dt neutrons (14 MeV) and 100% of this 14 MeV gets converted to reactor power (which is very far from the truth, but it's an assumption we're living with). Under this assumption it converts the reactor power back into number of neutrons.

Reply:
Once we have a different neutron source set up it would be sensible to remove this "reactor power" variable and rely on the d and t densities instead.